uwblueprint/childrens-aid-society
# Intake Form Validation - Phone Numbers

## Intake Form Validation

[Intake Form Validation](https://www.notion.so/uwblueprintexecs/Intake-Form-Validation-5f93e04364a84eae9330c43a24c19031#26b6fe8a834241099e41747833d46370)



## Implementation description
* Added validation for phone numbers. The form accepts phone numbers of the following types: 
* Regular phone numbers - (XXX) XXX-XXXX
* International codes - +XXX (XXX) XXX-XXXX
* Extensions - (XXX) XXX-XXXX Ext. XXX...
* The parenthesis for the first three numbers is optional, the international codes are optional, all spacing and dashes are optional, the e in ext can be upper or lower case and the dot after the extension is optional 
*  Added validation for dates. The form accepts dates of the following types: 
* DD/MM/YYYY
* YYYY-M-DD



## Steps to test
1. On the landing page, create a new case 
2. Under case referral in REFERRING WORKER CONTACT, test that the above options work and that any other options are considered invalid 
3. Navigate to individual details using the next button and add a child 
4. Under the School / Daycare section, in SCHOOL/DAYCARE CONTACT INFORMATION, test that the above options work and that any other options are considered invalid 
7. Click on the Providers tab, add a new provider
8. Test the PRIMARY PHONE NUMBER and SECONDARY PHONE NUMBER using the above options (note that SECONDARY PHONE NUMBER is optional) 
9. Back to individual details, add visiting family
10. Test the PRIMARY PHONE NUMBER and SECONDARY PHONE NUMBER using the above options (note that SECONDARY PHONE NUMBER is optional) 
11. In the program details section of the intake form, test suggested start date.
12. Any form buttons should be disabled on incorrect input
Examples:
<img width="528" alt="image" src="https://github.com/uwblueprint/childrens-aid-society/assets/68048010/cec3c6a2-7287-4db3-8e9b-1c8c336da05b">

<img width="529" alt="image" src="https://github.com/uwblueprint/childrens-aid-society/assets/68048010/57344809-f971-48b8-a02c-9b9a737bb1d2">

<img width="1238" alt="Screen Shot 2024-02-21 at 9 34 17 PM" src="https://github.com/uwblueprint/childrens-aid-society/assets/108363018/0567bb5d-f5ef-4bd8-808b-f3ecca64e233">

<img width="1228" alt="Screen Shot 2024-02-21 at 9 34 39 PM" src="https://github.com/uwblueprint/childrens-aid-society/assets/108363018/5770d6c9-81a5-4c9e-8976-f50a43391ed7">

<img width="1245" alt="Screen Shot 2024-02-21 at 9 35 15 PM" src="https://github.com/uwblueprint/childrens-aid-society/assets/108363018/00b09722-fa56-478f-90d5-b21e2e0d0abd">


## What should reviewers focus on?
* Should we disable the "next" buttons if the input is wrong?

## Checklist
- [ x] My PR name is descriptive and in imperative tense
- [ x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ x] I have run the appropriate linter(s)
- [ x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
